### ISSUE TITLE
Support for env variables in application.properties

### ISSUE THREAD

**Issue by Cipher358** [smallrye/smallrye-reactive-messaging#419]
I have a use case where I'm using Quarkus with the intent of having a Kafka consumer.
```
@Incoming("events")
public void processEvent(int event)
```
### Background
In application.properties, I have the following:

`mp.messaging.incoming.events.topic=${PREFIX}.events`

When the topic name is extracted inside `ConnectorConfig:36`, `${PREFIX}` is treated as a literal String, while I'd like it to be replaced by an environment variable:
```
this.topic = overall.getOptionalValue(channelKey("topic"), String.class).orElse(null);
```
### Alternative
I was able to obtain the desired behaviour by using JAVA OPTS instead of application.properties, but this solution doesn't fit in with the rest of the system.

`java -Dmp.messaging.incoming.events.topic=dev.events -jar service/build/my-app-runner.jar`

### What I used

- `io.quarkus:quarkus-smallrye-reactive-messaging-kafka:1.1.0.Final`
- `io.smallrye.reactive:smallrye-reactive-messaging-provider:1.0.8`
- this guide as documentation: https://quarkus.io/guides/config

### Question
Is there an approach that could solve my issue? 
Otherwise, would it be ok to make a pull request where this functionality is included?

**Comment by kenfinnigan** [comment]
I think that environment variable parsing is present within Quarkus, but it may be missing for MP properties in Quarkus.

Have you raised an issue with Quarkus, as it's possible the fix is actually needed there instead?

**Comment by Cipher358** [comment]
Hi Ken, thanks for the quick reply. I have not raised it with Quarkus.

After debugging the startup process some more, I found the following line -- `io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory:74`

```
this.config = ConfigProvider.getConfig();
```

I set a breakpoint on it and captured the state of these 2 following evaluations.

<img width="1057" alt="Screenshot 2020-02-14 at 19 16 03" src="https://user-images.githubusercontent.com/13125782/74560484-90052d80-4f5e-11ea-8be5-b122158fff64.png">

`ConfigProviderResolver.instance().getConfig()` seems to be the config object where the env variables have been processed.

Any thoughts?

**Comment by kenfinnigan** [comment]
Inside `ConfiguredChannelFactory` what's the result of running `config.getOptionalValue("mp.messaging.incoming.events.topic")`?

It seems odd for `config` to be passed in and cleared if it isn't satisfied, and ignored if it is satisfied.

It almost seems like the if logic is backward.

**Comment by Cipher358** [comment]
<img width="1676" alt="Screenshot 2020-02-14 at 19 35 31" src="https://user-images.githubusercontent.com/13125782/74561923-85986300-4f61-11ea-8d91-4d6899b5ef38.png">

In the constructor of ConfiguredChannelFactory, it always goes in the else branch, so it's not using the injected config.
```
        if (config.isUnsatisfied()) {
            this.incomingConnectorFactories = null;
            this.outgoingConnectorFactories = null;
            this.config = null;
        } else {
            this.incomingConnectorFactories = incomingConnectorFactories;
            this.outgoingConnectorFactories = outgoingConnectorFactories;
            if (logConnectors) {
                LOGGER.info("Found incoming connectors: {}", getConnectors(beanManager, IncomingConnectorFactory.class));
                LOGGER.info("Found outgoing connectors: {}", getConnectors(beanManager, OutgoingConnectorFactory.class));
            }
            this.config = ConfigProvider.getConfig();
        }
```

**Comment by kenfinnigan** [comment]
The image doesn't seem to match with what you're describing, so I'm a bit confused.

If you breakpoint here: https://github.com/smallrye/smallrye-reactive-messaging/blob/master/smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/impl/ConfiguredChannelFactory.java#L66

What does `config.getOptionalValue("mp.messaging.incoming.events.topic")` return? `config` is injected into the constructor, so it would be curious to know what it returns

**Comment by Cipher358** [comment]
With a breakpoint on 66, `config.get().getOptionalValue("mp.messaging.incoming.events.topic", String.class);` evaluates to `dev.events`.

**Comment by kenfinnigan** [comment]
If you wouldn't mind proposing a PR to fix that would be great

**Comment by Cipher358** [comment]
I can happily do that, but it will have to be tomorrow. It's 8 pm local time :)

Thanks for the help!

**Comment by kenfinnigan** [comment]
Completely understand. Whenever you can provide one is greatly appreciated, no rush at all

**Comment by cescoffier** [comment]
Fixed with #422 

Thanks!
